x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/ProfilingPlugin.java

```diff
@@ -108,10 +111,12 @@ public Collection<Object> createComponents(
             registry.get().initialize();
             indexManager.get().initialize();
             dataStreamManager.get().initialize();
-            return List.of(registry.get(), indexManager.get(), dataStreamManager.get());
-        } else {
-            return Collections.emptyList();
         }
+        return Collections.singletonList(createLicenseChecker());
```

Comment:
See https://github.com/elastic/elasticsearch/pull/97546#discussion_r1339564573 for context on why we are returning the license checker in all cases (but no other components).